src/InsetOrderOptimizer.cpp

```diff
@@ -570,60 +565,23 @@ bool InsetOrderOptimizer::processInsetsWithOptimizedOrdering()
     added_something = false;
     const unsigned int num_insets = part.insets.size();
 
-    // if overlap compensation is enabled, gather all the level 0 and/or level X walls together
-    // and initialise the respective overlap computers
-    // NOTE: this code assumes that the overlap computers do not alter the order or number of the polys!
-    Polygons wall_0_polys;
-    if (mesh.settings.get<bool>("travel_compensate_overlapping_walls_0_enabled"))
-    {
-        wall_0_polys = part.insets[0];
-        wall_overlapper_0 = new WallOverlapComputation(wall_0_polys, mesh_config.inset0_config.getLineWidth());
-    }
-
-    Polygons wall_x_polys;
-    if (mesh.settings.get<bool>("travel_compensate_overlapping_walls_x_enabled"))
-    {
-        for (unsigned int inset_level = 1; inset_level < num_insets; ++inset_level)
-        {
-            wall_x_polys.add(part.insets[inset_level]);
-        }
-        // use a slightly reduced line width so that compensation only occurs between insets at the same level (and not between insets in adjacent levels)
-        wall_overlapper_x = new WallOverlapComputation(wall_x_polys, mesh_config.insetX_config.getLineWidth() - 1);
-    }
-
     // create a vector of vectors containing all the inset polys
     inset_polys.clear();
 
-    // if overlap compensation is enabled, use the polys that have been tweaked by the
-    // overlap computers, otherwise use the original, un-compensated, polys
+    // Use the original, un-compensated, polys
 
     inset_polys.emplace_back();
     for (unsigned int poly_idx = 0; poly_idx < part.insets[0].size(); ++poly_idx)
     {
-        if (wall_overlapper_0)
-        {
-            inset_polys[0].push_back(wall_0_polys[poly_idx]);
-        }
-        else
-        {
-            inset_polys[0].push_back(part.insets[0][poly_idx]);
-        }
+        inset_polys[0].push_back(part.insets[0][poly_idx]);
     }
 
-    unsigned int wall_x_polys_index = 0;
     for (unsigned int inset_level = 1; inset_level < num_insets; ++inset_level)
     {
         inset_polys.emplace_back();
         for (unsigned int poly_idx = 0; poly_idx < part.insets[inset_level].size(); ++poly_idx)
         {
-            if (wall_overlapper_x)
-            {
-                inset_polys[inset_level].push_back(wall_x_polys[wall_x_polys_index++]);
-            }
-            else
-            {
-                inset_polys[inset_level].push_back(part.insets[inset_level][poly_idx]);
-            }
+            inset_polys[inset_level].push_back(part.insets[inset_level][poly_idx]);
         }
     }
```

Comment:
And here the same. Maybe we can even move the outer wall case in with the rest of this `for` loop, letting `inset_level` start at 0. We don't have a difference any more between `wall_overlapper_x` and `wall_overlapper_0`, so these are now the same.

Reply:
With some luck, if the polygons are not being modified here, we could also just use `part.insets` and leave out the entire `inset_polys` variable. That would prevent a copy of the data.

Reply:
I merged the for loops into one. The inset_polys is modified along the way, so I don't think it's a good idea to replace it with the part.insets.